src/index.js

```diff
@@ -0,0 +1,168 @@
+import axios from 'axios';
+import execa from 'execa';
+import inquirer from 'inquirer';
+import os from 'os';
+import signale, { Signale } from 'signale';
+
+import { execute, executeShell } from './execute';
+import * as packageBundles from './packages';
+import { brewInstall } from './packages/installers';
+import foundInPath from './foundInPath';
+
+const log = new Signale({ interactive: true });
+
+const die = msg => {
+  signale.fatal(msg);
+  process.exit(1);
+};
+
+async function main() {
+  // Ensure running on macOS
+  if (os.type() !== 'Darwin') die('This script was written for macOS');
+
+  await installCLITools();
+
+  const requirements = [
+    { install: installHomebrew, name: 'brew' },
+    { check: hasBrewCask, install: installHomebrewCask, name: 'brew-cask' },
+    { force: true, install: setupN, name: 'n (folder setup)' },
+    { install: brewInstall('n'), name: 'n' },
+    { install: installNode, name: 'node' },
+    { install: brewInstall('yarn', '--without-node'), name: 'yarn' }
+  ];
+
+  // Run in series
+  for (const requirement of requirements) {
+    await installRequirement(requirement);
+  }
+
+  await promptInstallPackages(packageBundles);
+
+  log.star("You're good to go!");
+}
+
+async function hasBrewCask() {
+  try {
+    await execa('brew', ['cask', '--version']);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+async function installCLITools() {
+  log.await('Checking for macOS command line tools');
+
+  try {
+    await execa('xcode-select', ['-p']);
+    log.success('macOS command line tools found');
+  } catch (e) {
+    log.await('Installing for macOS command line tools');
+    await execa('xcode-select', ['--install']);
+    log.success('Installed macOS command line tools');
+  }
+}
+
+async function installHomebrew() {
+  const response = await axios.get(
+    'https://raw.githubusercontent.com/Homebrew/install/master/install'
+  );
+
+  const installScript = response.data;
+
+  await execute('ruby', ['-e', installScript]);
+
+  return executeShell(`
+    # Ensure folder for n exists
```

Comment:
I admit I cheat a bit and use `executeShell()`.  While doing this in Node is possible, I think it's just too much for what it does.